Pokedex/Pokedex/View Controllers/PokemonDetailViewController.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  PokemonDetailViewController.swift
+//  Pokedex
+//
+//  Created by Moses Robinson on 1/25/19.
+//  Copyright © 2019 Moses Robinson. All rights reserved.
+//
+
+import UIKit
+
+class PokemonDetailViewController: UIViewController {
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        updateViews()
+        imageView.isHidden = true
+    }
+    
+    private func updateViews() {
+        guard let pokemon = pokemon else { return }
+        
+        nameLabel.text = pokemon.name.capitalized
+        idLabel.text = "ID: \(pokemon.id)"
+        
+        let typeString = pokemon.types.map {$0.type.name.capitalized}.joined(separator: "/")
+        typeLabel.text = "Types: \(typeString)"
+        
+        let abilityString = pokemon.abilities.map {$0.ability.name.capitalized}.joined(separator: ", ")
+        abilityLabel.text = "Abilities: \(abilityString)"
+    
+        pokemonController?.fetchImage(for: pokemon, completion: { (data) in
+            guard let data = data else { return }
+            self.imageView.isHidden = false
+            let image = UIImage(data: data)
+            self.imageView.image = image
```

Comment:
`URLSession` is doing its work in background queue and if you need to update any UI, it should be done in main queue
```suggestion
            DispatchQueue.main.async {
                self.imageView.isHidden = false
                let image = UIImage(data: data)
                self.imageView.image = image
            }
```